packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/updater/ContainmentUpdater.java

```diff
@@ -86,17 +86,31 @@ public void update(IContainerLayoutData container) {
 
     private boolean shouldUpdate(IContainerLayoutData container) {
         boolean shouldUpdate = true;
+        boolean isLabelContained = this.isLabelContained(container);
+        if (isLabelContained && this.isLabelLargerThanNode(container)) {
+            shouldUpdate = true;
+        } else {
+            if (container instanceof NodeLayoutData) {
+                NodeLayoutData nodeLayoutData = (NodeLayoutData) container;
+                shouldUpdate = !(nodeLayoutData.isResizedByUser() && this.hasEveryChildWithinItsBounds(nodeLayoutData));
+            }
 
-        if (container instanceof NodeLayoutData) {
-            NodeLayoutData nodeLayoutData = (NodeLayoutData) container;
-            shouldUpdate = !(nodeLayoutData.isResizedByUser() && this.hasEveryChildWithinItsBounds(nodeLayoutData));
+            shouldUpdate = shouldUpdate && (!container.getChildrenNodes().isEmpty() || isLabelContained);
```

Comment:
I barely understand when the node should be updated or not. I think the `if .. then ..` can be flattened.

Reply:
Well no matter what we do, this is barely understandable. Let's keep it that way.